app/javascript/components/retirement-form/index.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import moment from 'moment';
+
+import MiqFormRenderer from '@@ddf';
+import createSchema from './retirement-form.schema';
+import handleFailure from '../../helpers/handle-failure';
+
+const RetirementForm = ({ retirementID }) => {
+  const id = retirementID.replace(/\D/g,'');
+  const [{ initialValues, isLoading }, setState] = useState({
+    isLoading: !!id,
+    initialValues: { formMode: '' },
+  });
+
+  const cancelUrl = `/${ManageIQ.controller}/retire?button=cancel`;
+  const saveURL = `/${ManageIQ.controller}/retire?button=save`;
+
+  const onSubmit = (data) => {
+    miqSparkleOn();
+    const date = data.formMode === 'delay' ? moment().add({ days:Number(data.days), weeks:Number(data.weeks), months:Number(data.months), hours:Number(data.hours) })._d : data.retirementDate;
+    const request = API.post(`/api/services/${id}`, { action: 'request_retire', resource: { date, warn: data.retirementWarning } });
+    request.then(() => {
+      miqAjaxButton(saveURL, { retire_date: date, retire_warn: data.retirementWarning });
+    }).catch(miqSparkleOff);
+  };
+
+  useEffect(() => {
+    if (id) {
+      API.get(`/api/services/${id}?attributes=retires_on,retirement_warn`).then((res) => {
+        if (res.retires_on != null) {
+          setState({
+            isLoading: false,
+            initialValues: {
+              retirementDate: res.retires_on,
+              retirementWarning: res.retirement_warn || '',
+              formMode: 'date',
```

Comment:
You should be passing non-API pulled `initialValues` in the schema as `initialValue: 'date'` for this given entity.